resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/dtree/DeltaDataTree.java

```diff
@@ -909,6 +913,14 @@ public void setData(IPath key, Object data) {
 		assembleNode(key, new DataDeltaNode(key.lastSegment(), data));
 	}
 
+	public void setRootData(Object data) {
+		setData(rootKey(), data);
+	}
+
+	public Object getRootData() {
```

Comment:
Is this really an Object? I see below explicit cast to ElementTree

Reply:
the ElementTree only uses it as ElementTree, but there are other Trees that use it differently :-(
org.eclipse.core.internal.dtree.DeltaDataTree.basicCompare(DeltaDataTree, IComparator, IPath) and compareWith set NodeComparison as rootData